tfjs-backend-webgpu/src/shader_preprocessor.ts

```diff
@@ -128,19 +129,19 @@ export function makeShader(
 
     // Only used when the y/z dimension of workgroup size is 1.
     fn getGlobalIndex() -> i32 {
-      if (numWorkgroups.y == 1u && numWorkgroups.z == 1u) {
-        return i32(globalId.x);
-      }
-
-      let localInvocationIndex = localId.z * workGroupSizeX * workGroupSizeY +
-          localId.y * workGroupSizeX + localId.x;
-      let workGroupID = (globalId - localId)/vec3<u32>(
-          workGroupSizeX, workGroupSizeY, workGroupSizeZ);
-
-      return i32((workGroupID.z * numWorkgroups.x * numWorkgroups.y +
-        workGroupID.y * numWorkgroups.x + workGroupID.x) *
-        (workGroupSizeX * workGroupSizeY * workGroupSizeZ) +
-        localInvocationIndex);
+      ${
+      program.dispatch[1] === 1 && program.dispatch[2] === 1 ?
```

Comment:
`program.dispatch[1] === 1 && program.dispatch[2] === 1` should be part of the shader key in `makeShaderKey`?